crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -424,3 +422,17 @@ impl AcirVarData {
 /// A Reference to an `AcirVarData`
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub(crate) struct AcirVar(usize);
+
+#[test]
+fn repeat_op() {
+    let mut ctx = AcirContext::default();
+    let var1 = ctx.add_variable();
+    let var2 = ctx.add_variable();
+    let mul = ctx.mul_var(var1, var2);
+    let should_be_same_mul = ctx.mul_var(var1, var2);
+    // It's only on the third attempt that `add_data`'s `data_reverse_map` & `data` equal length
+    // assert previously failed (prior to adding a dupe check inside of add_data).
+    let should_be_same_mul_again = ctx.mul_var(var1, var2);
```

Comment:
why do you need to do mul_var again?

Reply:
I had used it for the 1-bit optimised XOR, which I've since removed. I'm still using it for OR, but I'm yet to trigger the corruption via that usage...

Reply:
So could this test be simpler by removing `let should_be_same_mul_again = ctx.mul_var(var1, var2);` ?

Reply:
Removing it also makes the test fail without the fix -- will push